donation-web/src/main/java/com/donation/web/VideoStreamingController.java

```diff
@@ -0,0 +1,110 @@
+package com.donation.web;
+
+import com.donation.crud.VideoCrudService;
+import com.donation.donor.model.Video;
+import com.donation.web.crud.Assets;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+import reactor.core.publisher.Flux;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+@RestController
+@RequestMapping("/api/stream")
+public class VideoStreamingController {
+
+    @Autowired
+    private VideoCrudService videoCrudService;
+
+    private List<Video> videos = new ArrayList<Video>();
+
+    @GetMapping("/{idProject}")
+    public StreamingResponseBody stream(@PathVariable Long idProject) throws FileNotFoundException {
+        Random random = new Random();
+        videos = videoCrudService.getVideosByProject(idProject);
```

Comment:
please move 'select Random video' to core, it's a business logic that could be changed later to use some algorithms (machine learning) to define the video to show based on user data...